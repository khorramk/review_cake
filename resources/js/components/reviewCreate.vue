<template>
        <form  v-on:submit.prevent="reviewCreation"  class="create-form-reviews form-container">
            <label class="create-form-reviews__review-creation__label" for="review_creation">write your reviews</label>
            <textarea class="create-form-reviews__review-creation__body" v-model="reviewBody" id="" cols="30" rows="10"></textarea>
            <input :disabled="$store.state.isDisable" class="create-form-reviews__review-creation__submit-btn"  type="submit" value="add your review">
        </form>
</template>
<script>
export default {
   data() {
       return {
           reviewBody: ''
       }
   },
   methods: {
        //method for review creation
    reviewCreation(){
        this.$store.commit('disableButton');
        axios.post('/api/reviews',{
            reviewCreation: this.reviewBody,
            })
            .then(()=> {
                window.location.href = '/reviews';
            })
            .catch((err)=> console.log(err));
    }
   },
};
</script>

