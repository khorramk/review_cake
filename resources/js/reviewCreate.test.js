

import Vue from 'vue';
import axios from 'axios';
import VueRouter from 'vue-router';
import reviewCreate from './components/reviewCreate';

const vm = new Vue(reviewCreate);


it('renders correct elements', () => {
    
    vm.$nextTick().then(()=> {
 
     expect(vm).toMatchSnapshot();
    })
  })