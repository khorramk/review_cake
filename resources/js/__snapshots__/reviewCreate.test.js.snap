// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct elements 1`] = `
Vue {
  "$attrs": Object {},
  "$children": Array [],
  "$createElement": [Function],
  "$listeners": Object {},
  "$options": Object {
    "_base": [Function],
    "components": Object {},
    "data": [Function],
    "directives": Object {},
    "filters": Object {},
    "methods": Object {
      "reviewCreation": [Function],
    },
    "render": [Function],
    "staticRenderFns": Array [],
  },
  "$parent": undefined,
  "$refs": Object {},
  "$root": [Circular],
  "$scopedSlots": Object {},
  "$slots": Object {},
  "$vnode": undefined,
  "_c": [Function],
  "_data": Object {
    "reviewBody": "",
  },
  "_directInactive": false,
  "_events": Object {},
  "_hasHookEvent": false,
  "_inactive": null,
  "_isBeingDestroyed": false,
  "_isDestroyed": false,
  "_isMounted": false,
  "_isVue": true,
  "_renderProxy": Vue {
    "$attrs": Object {},
    "$children": Array [],
    "$createElement": [Function],
    "$listeners": Object {},
    "$options": Object {
      "_base": [Function],
      "components": Object {},
      "data": [Function],
      "directives": Object {},
      "filters": Object {},
      "methods": Object {
        "reviewCreation": [Function],
      },
      "render": [Function],
      "staticRenderFns": Array [],
    },
    "$parent": undefined,
    "$refs": Object {},
    "$root": [Circular],
    "$scopedSlots": Object {},
    "$slots": Object {},
    "$vnode": undefined,
    "_c": [Function],
    "_data": Object {
      "reviewBody": "",
    },
    "_directInactive": false,
    "_events": Object {},
    "_hasHookEvent": false,
    "_inactive": null,
    "_isBeingDestroyed": false,
    "_isDestroyed": false,
    "_isMounted": false,
    "_isVue": true,
    "_renderProxy": [Circular],
    "_self": [Circular],
    "_staticTrees": null,
    "_uid": 0,
    "_vnode": null,
    "_watcher": null,
    "_watchers": Array [],
    "reviewBody": "",
    "reviewCreation": [Function],
  },
  "_self": [Circular],
  "_staticTrees": null,
  "_uid": 0,
  "_vnode": null,
  "_watcher": null,
  "_watchers": Array [],
  "reviewBody": "",
  "reviewCreation": [Function],
}
`;
